src/extension.ts

```diff
@@ -88,12 +86,20 @@ export async function activate(context: vscode.ExtensionContext) {
 		},
 	));
 
-	if (activeTextEditor !== undefined) {
+	// decorate the active editor now
+	const activeTextEditor = vscode.window.activeTextEditor;
+	if (!activeTextEditor || isNotArbFile(activeTextEditor.document)) {
+		return;
+	} else {
+		let [messageList, errors] = parser.parse(activeTextEditor.document.getText())!;
+		commonMessageList = messageList;
 		decorator.decorate(activeTextEditor, messageList);
 		diagnostics.diagnose(activeTextEditor, messageList, errors);
```

Comment:
This code is repeated in quite a few places, it may be worth extracting it to a function that takes an Editor, parses it, captures the last parse results, and adds decorations/diagnostics?